core/src/main/scala/akka/persistence/jdbc/state/scaladsl/DurableStateStore.scala

```diff
@@ -0,0 +1,51 @@
+package akka.persistence.jdbc.state.scaladsl
+
+import scala.concurrent.{ ExecutionContext, Future }
+import slick.jdbc.{ JdbcBackend, JdbcProfile }
+import akka.Done
+import akka.persistence.state.scaladsl.{ DurableStateUpdateStore, GetObjectResult }
+import akka.serialization.Serialization
+import akka.persistence.jdbc.state.{ AkkaSerialization, DurableStateQueries }
+import akka.persistence.jdbc.config.DurableStateTableConfiguration
+import akka.persistence.jdbc.state.DurableStateTables
+import akka.dispatch.ExecutionContexts
+
+class DurableStateStore[A](
+    db: JdbcBackend#Database,
+    profile: JdbcProfile,
+    durableStateConfig: DurableStateTableConfiguration,
+    serialization: Serialization)(implicit ec: ExecutionContext)
+    extends DurableStateUpdateStore[A] {
+  import profile.api._
+
+  val queries = new DurableStateQueries(profile, durableStateConfig)
+
+  def getObject(persistenceId: String): Future[GetObjectResult[A]] =
+    db.run(queries._selectByPersistenceId(persistenceId).result).map { rows =>
+      rows.headOption match {
+        case Some(row) =>
+          GetObjectResult(AkkaSerialization.fromRow(serialization)(row).toOption.asInstanceOf[Option[A]], row.seqNumber)
+
+        case None =>
+          throw new Exception(s"State object creation failed during fetch from store: persistenceId $persistenceId")
```

Comment:
This shouldn't throw. It should be None in the `GetObjectResult` value. In that case you can use 0 as seqNr.

Reply:
ok, will change ..